### PR TITLE
[AUTOMATE-2138] Always refetch latest data in UI for projects, policies, and roles

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

When calls to `GET` and `GET_ALL` happen in the store, clear the store of the object in question and refetch so that we always have the latest data.

Also updated all the subscriptions to filter until the fetch is completed to avoid flickering and other undesirable side effects of hitting the subscribe code multiple times.

### :athletic_shoe: How to Build and Test the Change

Spin up UI. Go to a role, policy, and project from both the list view and by refreshing from the details page. Everything should function correctly.

### :white_check_mark: Checklist

- [ ] Tests added/updated?
- [ ] Docs added/updated?

### CODE REVIEW

File: components/automate-ui/src/app/pages/project/details/project-details.component.ts [hunk 1]
```diff
@@ -85,28 +95,25 @@ export class ProjectDetailsComponent implements OnInit, OnDestroy {
...
-        });
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(getAllRulesForProjectStatus),
+      this.store.select(projectFromRoute),
+      this.store.select(allRules)
+    ]).pipe(
+        filter(([getProjectSt, getRulesSt, _projectState, _allRulesState]) =>
+          getProjectSt === EntityStatus.loadingSuccess &&
+          getRulesSt === EntityStatus.loadingSuccess),
```
Comment (msorens) [inline components/automate-ui/src/app/pages/project/details/project-details.component.ts]: Wondering if this could be split into having a `loading$` Observable in its own `combineLatest` as Lance did in his recent PR, then you could leverage the new `allLoadedSuccessfully` function he is introducing. See https://github.com/chef/automate/pull/2151.
I bring that up because I think it would then make this a bit simpler.

File: components/automate-ui/src/app/pages/policy/details/policy-details.component.ts [hunk 2]
```diff
@@ -70,9 +75,7 @@ export class PolicyDetailsComponent implements OnInit, OnDestroy {
           }
         });
         delete this.policy.members;
-        return members;
-      }));
-    this.members$.subscribe();
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/policy/details/policy-details.component.ts]: Thank you for cleaning this up. I keep seeing this but thought it was out of the scope of my work.

File: components/automate-ui/src/app/pages/policy/details/policy-details.component.html [hunk 3]
```diff
@@ -36,15 +36,15 @@
       </section>
     </section>
     <section class="page-body" *ngIf="tabValue === 'members'">
-      <div *ngIf="(members$ | async)?.length === 0" class="empty-state">
+      <div *ngIf="members.length === 0" class="empty-state">
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/pages/policy/details/policy-details.component.html]: I've been wondering if it's better to use `members` or observable of `members` in the template. Why did you change it here?

File: components/automate-ui/src/app/pages/policy/details/policy-details.component.ts [hunk 4]
```diff
@@ -51,16 +52,20 @@ export class PolicyDetailsComponent implements OnInit, OnDestroy {
...
-    this.members$ = this.store.select(policyFromRoute).pipe(
-      filter(identity),
-      takeUntil(this.isDestroyed),
-      map((state) => {
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(policyFromRoute)
+    ]).pipe(
+      filter(([status, _]) => status === EntityStatus.loadingSuccess),
+      filter(([_, state]) => !isNil(state)),
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-ui/src/app/pages/policy/details/policy-details.component.ts]: I think the intention is more clear as separate filters but I don't have a strong opinion

File: components/automate-ui/src/app/pages/policy/details/policy-details.component.html (hunk 3 above)
Comment (tylercloke) [reply components/automate-ui/src/app/pages/policy/details/policy-details.component.html]: In all the other places I've seen we just initialize to an empty array and then populate after the subscribe finishes. Doesn't look to me that the complexity of having it be an obserable gives us anything? Switching it to a plain old array just simplified the templates and made the ts code more standardized.

File: components/automate-ui/src/app/pages/project/details/project-details.component.ts (hunk 1 above)
Comment (tylercloke) [reply components/automate-ui/src/app/pages/project/details/project-details.component.ts]: I added a `allLoaded` function but it wasn't obvious to me how to easily move `isLoading` to an observable since it's being used in typescript functions. Can make a follow up PR if this is easy to fix.

**Comment by tylercloke** [comment]
@bcmdarroch fixed
